File: front-end/ebanking-app/src/app/core/accounts/account.ts
export class Account {
    constructor(public id: string,
                public userId: string,
                public accountName: string,
                public accountNumber: string,
                public accountType: string, // TODO change to enum
                public balance: number,
                public availableBalance: number,
                public interestRate: number,
                public bin?: string,
                public bsbCode?: string,
                public accountNickName?: string) {
    }

    displayName(): string {
        return `${this.accountName} - ${this.accountType}`;
    }
}

File: gateway/src/main/java/onlinebank/gateway/model/Account.java
package onlinebank.gateway.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Account {
    private String id;
    private String userId;
    private String accountNumber;
    private String accountName;
    private String bin;
    private String bsbCode;
    private long balance;
    private long availableBalance;
    private String accountType;
    private double interestRate;
}
